cross-media-measurement/src/main/cc/wfa/measurement/common/crypto/liquid_legions_v2_encryption_utility.cc

```diff
@@ -39,11 +39,15 @@ namespace {
 // to this group, i.e., having the same blinded register index.
 absl::Status MergeCountsUsingSameKeyAggregation(
     absl::Span<const size_t> sub_permutation, absl::string_view registers,
-    ProtocolCryptor& protocol_cryptor, std::string& response) {
+    ProtocolCryptor& protocol_cryptor, int total_sketches_count,
+    std::string& response) {
   if (sub_permutation.empty()) {
     return absl::InternalError("Empty sub permutation.");
   }
-
+  if (sub_permutation.size() > total_sketches_count) {
```

Comment:
Would it make sense to push this check to the caller?

Reply:
no, there are multiple places calling this, that would be copying/pasting the code multiple times.